File: patterns/java/src/refactoring_guru/patterns/observer/example/Editor.java
package refactoring_guru.patterns.observer.example;

import java.io.File;

public class Editor {
    public EventManager events;
    private File file;

    public Editor() {
        this.events = new EventManager();
    }

    public void openFile(String filePath) {
        this.file = new File(filePath);
        events.execute("open", file);
    }

    public void saveFile() {
        this.file = new File("D:/tmp");
        events.execute("save", file);
    }
}

File: patterns/cs/src/RefactoringGuru/Observer/Example/Editor/Editor.cs
using System;
using System.IO;
using RefactoringGuru.Observer.Example.Publisher;

namespace RefactoringGuru.Observer.Example.Editor
{
    class Editor
    {
        public EventManager events;
        private FileInfo file;

        public Editor()
        {
            String[] args = { "open", "save" };
            this.events = new EventManager(args);
        }

        public void OpenFile(String filePath)
        {
            this.file = new FileInfo(filePath);
            events.Notify("open", file);
        }

        public void SaveFile()
        {
        if (this.file != null) {
                events.Notify("save", file);
            } else {
                throw new Exception("Please open a file first.");
            }
        }
    }
}
